DiagnosticsExtension/Controllers/ConnectionStringValidationController.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="UdpEchoTestController.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using DiagnosticsExtension.Models.ConnectionStringValidator;
+
+namespace DiagnosticsExtension.Controllers
+{
+    /// <summary>
+    /// Worker instances are running an udp echo server on port 30000. This controller is for checking the connection between target 
+    /// worker instance by pinging and checking the echoed result.
+    /// </summary>
+    [RoutePrefix("api/connectionstringvalidation")]
+    public class ConnectionStringValidationController : ApiController
+    {
+        [HttpGet]
+        [Route("test")]
+        public async Task<HttpResponseMessage> Test(string connStr, int? typeId = null)
+        {
+            var typeValidatorMap = new IConnectionStringValidator[]
+            {
+                new SqlServerValidator()
+            }.ToDictionary(v => v.Type, v => v);
+
+            if (typeId != null)
+            {
+                var enumType = (ConnectionStringType)typeId.Value;
+                if (typeValidatorMap.ContainsKey(enumType))
+                {
+                    var result = await typeValidatorMap[enumType].Test(connStr);
+                    return Request.CreateResponse(HttpStatusCode.OK, new { result, connStr });
+                }
+                else
+                {
+                    return Request.CreateErrorResponse(HttpStatusCode.NotFound, $"No supported validator found for typeId={typeId.Value}");
+                }
+            }
+            else
+            {
+                foreach (var p in typeValidatorMap)
+                {
+                    try
+                    {
+                        if (p.Value.IsValid(connStr))
+                        {
+                            var result = await p.Value.Test(connStr);
+                            return Request.CreateResponse(HttpStatusCode.OK, new { result, connStr });
+                        }
+                    }
+                    catch (Exception e)
+                    { 
+                    }
+                }
+                return Request.CreateErrorResponse(HttpStatusCode.NotFound, $"No supported validator found for provided connection string");
+            }
+        }
+
+        [HttpGet]
+        [Route("testappsetting")]
+        public async Task<HttpResponseMessage> TestAppSetting(string appSettingName, int? typeId = null)
+        {
+            var envDict = Environment.GetEnvironmentVariables();
```

Comment:
Does DaaS always have all the app settings that a WebApp/FunctionApp process has and are they updated each time the app settings are changed?

Reply:
I believe so, as the user program also reads these appsettings as system environment variables